src/js/components/Carousel.js

```diff
@@ -133,29 +140,42 @@ export default class Carousel extends Component {
       const { activeIndex } = this.state;
       const { intl } = this.context;
       const numSlides = children.length;
-
-      this.setState({
-        activeIndex: (activeIndex + 1) % numSlides
-      }, () => {
+      const index = (activeIndex + 1) % numSlides;
+      const announceFunc = () => {
         const slideNumber = Intl.getMessage(intl, 'Slide Number', {
           slideNumber: this.state.activeIndex + 1
         });
         const activatedMessage = Intl.getMessage(intl, 'Activated');
         announce(`${slideNumber} ${activatedMessage}`, 'polite');
-      });
+      };
 
+      if (! this.props.hasOwnProperty('activeIndex')) {
+        this.setState({
+          activeIndex: index
+        }, announceFunc );
+      }
       if (!infinite && activeIndex === numSlides - 1) {
         clearInterval(this._slideAnimation);
       }
+
+      if (this.props.onActive) {
+        this.props.onActive(index);
+        announceFunc();
```

Comment:
I don't think we need to call announceFunc here as we already call when setState is invoked.